providers/implementations/keymgmt/ec_kmgmt.c

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/core_numbers.h>
+#include <openssl/core_names.h>
+#include <openssl/bn.h>
+#include <openssl/ec.h>
+#include <openssl/objects.h>
+#include <openssl/params.h>
+#include "internal/param_build.h"
+#include "prov/implementations.h"
+#include "prov/providercommon.h"
+
+static OSSL_OP_keymgmt_importdomparams_fn ec_importdomparams;
+static OSSL_OP_keymgmt_exportdomparams_fn ec_exportdomparams;
+static OSSL_OP_keymgmt_importkey_fn ec_importkey;
+static OSSL_OP_keymgmt_exportkey_fn ec_exportkey;
+
+static ossl_inline
+EC_GROUP *params_to_domparams(const OSSL_PARAM params[])
+{
+    const OSSL_PARAM *param_ec_name;
+    EC_GROUP *ecg = NULL;
+    char *curve_name = NULL;
+    int ok = 0;
+
+    param_ec_name = OSSL_PARAM_locate_const(params, OSSL_PKEY_PARAM_EC_NAME);
+    if (param_ec_name == NULL) {
+        /* explicit parameters */
+
+        /*
+         * TODO(3.0): should we support explicit parameters curves?
+         */
+        return NULL;
+    } else {
+        /* named curve */
+        int curve_nid;
+
+        if (!OSSL_PARAM_get_utf8_string(param_ec_name, &curve_name, 0)
+                || curve_name == NULL
+                || (curve_nid = OBJ_sn2nid(curve_name)) == NID_undef)
+            goto err;
+
+        if ((ecg = EC_GROUP_new_by_curve_name(curve_nid)) == NULL)
```

Comment:
if fips there is no default libctx.
So you would need to use EC_GROUP_new_by_curve_name_ex(). There will be more of them. Maybe dont tackle FIPS for this PR but keep it in mind..

Reply:
Yes thanks! I already took notes about functions that will need to be replaced with the `*_ex()` version for FIPS